File: app/src/test/java/de/westnordost/streetcomplete/util/ReverseIteratorTest.java
package de.westnordost.streetcomplete.util;

import org.junit.Test;

import java.util.Arrays;

import static org.junit.Assert.*;

public class ReverseIteratorTest
{
	@Test public void reverse()
	{
		ReverseIterator<String> it = new ReverseIterator<>(Arrays.asList("a", "b", "c"));
		assertEquals("c", it.next());
		assertEquals("b", it.next());
		assertEquals("a", it.next());
		assertFalse(it.hasNext());
	}
}

File: app/src/test/java/de/westnordost/streetcomplete/util/ReverseIteratorTest.kt
package de.westnordost.streetcomplete.util

import org.junit.Assert.*
import org.junit.Test

class ReverseIteratorTest {

    @Test fun reverse() {
        val it = ReverseIterator(mutableListOf("a", "b", "c"))
        assertEquals("c", it.next())
        assertEquals("b", it.next())
        assertEquals("a", it.next())
        assertFalse(it.hasNext())
    }
}
